### PR TITLE
Add create/delete resourceserver commands

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/12941

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by corest** [comment]
Docs how it is used can be found here https://github.com/giantswarm/giantswarm/pull/13485/files#diff-825bd87a6a16775633127db598c5654dR24

File: pkg/auth0/auth0.go [hunk 1]
```diff
@@ -1,58 +1,164 @@
...
+		audience:    fmt.Sprintf(managementAudience, config.Tenant),
 
 		httpClient: httpClient,
 	}, nil
 }
 
-func readTokenFromFile(tenant string) (string, error) {
-	return "", nil
+func (a0 *Auth0) CreateResourceServer(rr *ResouceServer) (*ResouceServer, error) {
+	endpoint := fmt.Sprintf("%sresource-servers", a0.audience)
```
Comment (kopiczko) [inline pkg/auth0/auth0.go:77]: Doesn't need a slash? I mean `/sresource-servers`.

**Review by kopiczko — COMMENTED** [review]
Code-wise looks ok. I'm not sure about edge case situation handling. I also have no idea what are resource servers in auth0. But I guess we'll have some docs for installation creation.

File: cmd/create/resourceserver/flag.go [hunk 2]
```diff
@@ -7,21 +7,48 @@ import (
...
 
 func (f *flag) Init(cmd *cobra.Command) {
-	cmd.Flags().StringVar(&f.Name, flagName, "", `Name of the new release. Must follow semver format.`)
+	cmd.Flags().StringVar(&f.Name, flagName, "", `Name of a new resource server.`)
+	cmd.Flags().StringVar(&f.Identifier, flagIdentifier, "", `Resource server identifier.`)
+	cmd.Flags().StringVar(&f.Tenant, flagTenant, "giantswarm", `Auth0 tenant.`)
+	cmd.Flags().BoolVar(&f.AllowOfflineAccess, flagAllowOfflineAccess, true, `Allow offline access.`)
+	cmd.Flags().IntVar(&f.TokenLifetime, flagTokenLifetime, 300, `Token lifetime.`)
+	cmd.Flags().IntVar(&f.TokenLifetimeWeb, flagTokenLifetimeWeb, 300, `Token lifetime for web`)
+	cmd.Flags().StringVar(&f.SigningAlgorithm, flagSigningAlgorithm, "RS256", `Signing algorithm.`)
```
[the remark this answers is not in the document]
Comment (corest) [reply cmd/create/resourceserver/flag.go:36]: Maybe. I'm not sure. I could make them const, but also don't see issues with keeping it here

File: pkg/auth0/auth0.go (hunk 1 above)
Comment (corest) [reply pkg/auth0/auth0.go:77]: Updated. I had a `/` in endpoint address

File: cmd/create/resourceserver/runner.go [hunk 3]
```diff
@@ -33,5 +36,45 @@ func (r *runner) Run(cmd *cobra.Command, args []string) error {
...
+		}
+	}
+
+	resourceServerExists, err := a0.ResourceServerExists(r.flag.Identifier)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if resourceServerExists {
+		fmt.Printf("Resource server with %#q identifier already exists.\n", r.flag.Identifier)
```
[the remark this answers is not in the document]
Comment (corest) [reply cmd/create/resourceserver/runner.go:59]: I don't think so. If resource server exists - we just skip creation. Or do you mean implementing this inside pkg?

[the remark this answers is not in the document]
Comment (corest) [reply cmd/create/resourceserver/runner.go:59]: Moved into pkg

File: cmd/delete/resourceserver/runner.go [hunk 4]
```diff
@@ -33,5 +36,36 @@ func (r *runner) Run(cmd *cobra.Command, args []string) error {
...
+		}
+	}
+
+	resourceServerExists, err := a0.ResourceServerExists(r.flag.Identifier)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if !resourceServerExists {
+		fmt.Printf("Resource server with %#q identifier doesn't exist.\n", r.flag.Identifier)
```
[the remark this answers is not in the document]
Comment (corest) [reply cmd/delete/resourceserver/runner.go:59]: Delete operation is idempotent and always gives 204. So I've removed validation here

File: cmd/create/resourceserver/runner.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kopiczko) [reply cmd/create/resourceserver/runner.go:59]: But what if the resource has a different config?

**Review by kopiczko — APPROVED** [review]

**Pull request merged**